TS_1/task2.1.ts

```diff
@@ -0,0 +1,48 @@
+// 1. Создайте интерфейс IEmployee с полями name, surname, salary, address (типы для этих полей такие же как в ItEmployee из таск 1)
+//   Создайте функцию getEmployeeInfo(employee), выводящую в консоль всю информацию про employee (формат текста придумать самим)
+//   Создайте type guard isItEmployee, принимающий юнион типов IEmployee и ItEmployee. Используйте его в функции getEmployeeInfo. 
+//   Если на входе itEmployee - выводите в консоль все поля айтишника (а не только те, что в employee)
+//   Функция должна принимать union type между IEmployee и ItEmployee, и через тайпгвард определять что за объект и как работать с ним 
+
+interface IEmployee {
+    name: string;
+    surname: string;
+    readonly salary: number;
+    address?: IAddress;
+    projectNames: string[];
+}
+ 
+function isItEmployee(obj: object): obj is ItEmployee {
+    return "ItEmployee" in obj;
+}
```

Comment:
No, it's not a valid typeGuard. 

- There is no "ItEmployee" key
- obj: object is not a valid typization